internal/transport/http2_server.go

```diff
@@ -375,16 +375,39 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		}
 	}
 
-	if !isGRPC || headerError {
+	if frame.StreamEnded() && !isGRPC || headerError {
 		t.controlBuf.put(&cleanupStream{
 			streamID: streamID,
 			rst:      true,
-			rstCode:  http2.ErrCodeProtocol,
+			rstCode:  http2.ErrCodeNo,
 			onWrite:  func() {},
 		})
 		return false
 	}
 
+	if !isGRPC {
```

Comment:
> In the case of !isGRPC, I think the right response is an HTTP status (maybe 406 Not Acceptable(??)) in a HTTP/2 HEADERS frame with END_STREAM set.

> Bad content-type. HTTP 415 with a text/plain error message. Also a gRPC status of INTERNAL with same error message

Bad `content-type` - return a http 415 and end_stream

Reply:
> with a text/plain error message

Should we do this as well?  That would require a body (in a data frame), I believe.

> Also a gRPC status of INTERNAL with same error message

Can you include `grpc-status` and `grpc-message` headers as well?  Is there anything else that we need to include with our headers here?